tools/mkclass/classcompiler.cpp

```diff
@@ -376,19 +376,32 @@ void ClassCompiler::HandleClass(const Klass& klass, const ClassDebugInfo&)
 		<< "}" << std::endl << std::endl;
 
 	/* GetLoadDependencies */
-	m_Header << "\t" << "const std::unordered_set<Type*>& GetLoadDependencies() const override;" << std::endl;
+	if (!klass.LoadDependencies.empty()) {
+		m_Header << "\tconst std::unordered_set<Type*>& GetLoadDependencies() const override;" << std::endl;
 
-	m_Impl << "const std::unordered_set<Type*>& TypeImpl<" << klass.Name << ">::GetLoadDependencies() const" << std::endl
-		<< "{" << std::endl
-		<< "\t" << "static const std::unordered_set<Type*> deps ({" << std::endl;
+		m_Impl << "const std::unordered_set<Type*>& TypeImpl<" << klass.Name << ">::GetLoadDependencies() const" << std::endl
+			<< "{" << std::endl
+			<< "\tstatic const auto deps ([] {" << std::endl;
 
-	for (const std::string& dep : klass.LoadDependencies)
-		m_Impl << "\t\t" << "GetByName(\"" << dep << "\").get()," << std::endl;
+		for (auto& dep : klass.LoadDependencies)
+			m_Impl << "\t\tauto type" << dep << " (GetByName(\"" << dep << "\").get());" << std::endl;
```

Comment:
That's much better, but why are you using two different loops for declaring and verifying them? Can't you just combine them like that?
```suggestion
		for (auto& dep : klass.LoadDependencies) {
			m_Impl << "\t\tauto type" << dep << " (GetByName(\"" << dep << "\").get());" << std::endl;
			m_Impl << "\t\tVERIFY(type" << dep << ");" << std::endl;
		}

		m_Impl << std::endl << "\t\treturn std::unordered_set<Type*>{";
```

Reply:
But I prefer this:

```c++
const std::unordered_set<Type*>& TypeImpl<Service>::GetLoadDependencies() const
{
	static const auto deps ([] {
		auto typeApiListener (GetByName("ApiListener").get());
		auto typeEndpoint (GetByName("Endpoint").get());
		auto typeHost (GetByName("Host").get());
		auto typeZone (GetByName("Zone").get());

		VERIFY(typeApiListener);
		VERIFY(typeEndpoint);
		VERIFY(typeHost);
		VERIFY(typeZone);

		return std::unordered_set{ typeApiListener, typeEndpoint, typeHost, typeZone, };
	}());

	return deps;
}
```